### DISCUSSION TITLE
About ComputeSmearedCrackingStress

### DISCUSSION THREAD

**Discussion by Xelver** [idaholab/moose#20546] [Q&A Modules: Solid mechanics] [answered]
Hi,

I'd like to use `ComputeSmearedCrackingStress` with `AbruptSoftening` to compute the failure of ceramic materials. I tried to familiarize myself with this function through the test example  "cracking.i".  The mesh file used in this example was a 1x1x1 cube with single element and the failure stress was 1.68e6 Pa. 
When I tried to replace the mesh file with another one, which was also a 1x1x1 cube but with more elements, the failure stress decreased. The more elements, the more the stress dropped (The failure stress was 4.17e5 Pa for 1000 elements).  I'm confused, because normally the failure stress should not change.

**Comment by GiudGiud** [comment, marked as answer]
@jiangwen84

**Reply by GiudGiud** [reply]
@Xelver is this solved?

Is the failure stress converging as you refine the mesh?
Are you using the uniform refinement feature?
What value do you expect for the failure stress?

**Reply by Xelver** [reply]
Thanks for your reply.
Actually, I then have tried another way instead of `ComputeSmearedCrackingStress`. 

The failure stress is converging as the mesh refinement, and I'm using the uniform refinement.
From my point of view, the failure stress should remain the same as the mesh is refined (1.68e6 Pa).

**Reply by GiudGiud** [reply]
@dschwen thoughts on this?

**Reply by jiangwen84** [reply]
@Xelver Could you send us your input file with ``ComputeSmearedCrackingStress``? @bwspenc and I would like to take a look.

**Reply by Xelver** [reply]
OK, here's the input file. The output corresponding to the input file is as follow:
![607386f86c290c2a0cb9ed9bcab8c41](https://user-images.githubusercontent.com/74576082/168708687-3c735be1-511c-4d1a-bb70-fb84087460e9.png)
```#
# Simple pull test for cracking.
# The stress increases for two steps and then drops to zero.

[Mesh]
  type = GeneratedMesh
  dim = 3
  nx = 10
  ny = 10
  nz = 10
  #file = cracking_test.e
[]

[GlobalParams]
  displacements = 'disp_x disp_y disp_z'
[]

[Functions]
  [./displ]
    type = PiecewiseLinear
    x = '0 1 2 3  4'
    y = '0 1 0 -1 0'
  [../]
[]

[Modules/TensorMechanics/Master]
  [./all]
    strain = FINITE
    add_variables = true
    generate_output = 'stress_xx stress_yy stress_zz stress_xy stress_yz stress_zx strain_xx'
  [../]
[]

[BCs]
  [./pull]
    type = FunctionPresetBC
    variable = disp_x
    boundary = right #4
    function = displ
  [../]
  [./left]
    type = PresetBC
    variable = disp_x
    boundary = left#1
    value = 0.0
  [../]
  [./bottom]
    type = PresetBC
    variable = disp_y
    boundary = bottom#2
    value = 0.0
  [../]
  [./back]
    type = PresetBC
    variable = disp_z
    boundary = back#3
    value = 0.0
  [../]
[]

[Materials]
  [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 2.8e7
    poissons_ratio = 0
  [../]
  [./elastic_stress]
    type = ComputeSmearedCrackingStress
    cracking_stress = 1.68e6
    softening_models = abrupt_softening
  [../]
  [./abrupt_softening]
    type = AbruptSoftening
  [../]
[]

[Postprocessors]
  [./strain_xx]
    type = ElementAverageValue
    variable = strain_xx
    execute_on = 'initial TIMESTEP_END'
  [../]
  [./stress_xx]
    type = ElementAverageValue
    variable = stress_xx  #In this case, the maximum stress_xx is 4.168e5 at 0.015s.
    execute_on = 'initial TIMESTEP_END'
  [../]
[]

[Executioner]
  type = Transient

  petsc_options_iname = '-ksp_gmres_restart -pc_type -sub_pc_type'
  petsc_options_value = '101                asm      lu'

  line_search = 'none'

  l_max_its = 100
  nl_max_its = 100
  nl_rel_tol = 1e-8
  nl_abs_tol = 1e-8
  l_tol = 1e-5
  start_time = 0.0
  end_time = 0.1
  dt = 0.005
[]

[Outputs]
  exodus = true
[]
```

**Reply by GiudGiud** [reply]
any updates on this?
